aikit/models/ordinal.py

```diff
@@ -0,0 +1,364 @@
+                    raise TypeError("For multi-target classes should be a list, instead I got %s" % str(type(self.classes)))
+                
+                categories = self.classes
+
+        self._target_encoder = klass(categories=categories, dtype=np.int32)
+        
+        yd2 = convert_generic(make2dimensions(y), output_type=conversion_type)
+  
+        if conversion_type == DataTypes.NumpyArray and yd2.dtype.kind == 'U':
+            yd2 = yd2.astype(np.object, copy=False)
+        
+        y_encoded = self._target_encoder.fit_transform(yd2)
+
+        return y_encoded
+
+
+    @property
+    def classes_(self):
+        if self._mono_target:        
+            return self._target_encoder.categories_[0]
+        else:
+            return self._target_encoder.categories_
+
+class ClassifierFromRegressor(_BaseOrdinalClassifier):
+    """ this class transform a regressor into a classifier
+    it can be used for ordinal classification
+    
+    This model will transform the target into an increasing list of integer
+    and then fit a regression on that
+    """
+
+    def __init__(self,
+                 regressor,
+                 kernel_windows=0.2,
+                 classes="auto"
+                 ):
+        self.regressor=regressor
+        self.classes=classes
+        self.kernel_windows=kernel_windows
+
+
+    def fit(self, X, y):
+        
+        if not is_regressor(self.regressor):
+            raise TypeError("regressor should be a sklearn regressor")
+
+        y_encoded = self._prepare_target(y, klass=OrdinalEncoder, conversion_type=DataTypes.NumpyArray)
+
+        if self._mono_target:
+            y_int = y_encoded[:,0]
+            assert len(self._target_encoder.categories_) == 1
+        else:
+            y_int = y_encoded # I keep the multi-dimension            
+            assert len(self._target_encoder.categories_) == y.shape[1]
+
+        self.regressor.fit(X, y_int)
+        
+        return self
+
+
+    def predict(self, X):
+        y_hat = self.regressor.predict(X)     # call regressor
+        y_int_hat = (y_hat + 0.5).astype(np.int32)  # conversion to closest int
+        
+        y_hat = self._target_encoder.inverse_transform(make2dimensions(y_int_hat))        
+        
+        if self._mono_target:
+            y_hat = y_hat[:, 0]
+
+        return y_hat.astype(self._target_dtype)
+
+
+    def predict_proba(self, X):
+        y_hat = self.regressor.predict(X)     # call regressor
+        if self._mono_target:
+            y_hat_2d = y_hat[:, np.newaxis]
+            assert y_hat.ndim == 1
+        else:
+            y_hat_2d = y_hat
+            assert y_hat.ndim == 2
+            
+
+        pivot_integers = [np.arange(len(category))[np.newaxis, :] for category in self._target_encoder.categories_]
+        
+        distances_to_pivot = [np.abs(y_hat_2d[:, j:(j+1)] - pivot_integers[j]) for j in range(y_hat_2d.shape[1])] # Rmk [:, j:(j+1)]  so that I keep the dimension
```

Comment:
concatenate the two loops in one (by expanding a for loop or adding distance_to_proba in previous line ?

Reply:
done : one for loop